### DISCUSSION TITLE
Xiaomi RTCGQ11LM motion sensor does not expose temperature anymore?

### DISCUSSION THREAD

**Discussion by OttoL2** [Koenkk/zigbee2mqtt#12724] [Devices]
Hi all, relative noob here so apologies for any dumb questions... I have 5 RTCGQ11LM Xiaomi motion sensors working via Z2M in Home assistant, until a few days ago hardly ever had an issue with any of them for months (a few very occasional disconnects). I do read the temperature sensors in them to have a rough idea of room temperature, corrected them via the settings (specific) and it worked. Doesn't trigger any automations or the like, just for info for the moment. 
A day or so ago the temperature sensors all stopped working, showing as "not available" in HA. When I look in the (latest version auto-updated) Z2M dashboard, none of them exposes temperature anymore. "device temperature" now is exposed but I reckon that's the temperature of the built-in processor... Strange enough, the temperature is still shown, correctly, in "state" in the device's Z2M dashboards. It just is not provided anymore via MQTT. Is there anything I can change to the config to get the temps back? I know they're not precise etc. etc. but I really liked to have them there. As the hardware clearly allows it, is there any reason why temperature isn't exposed anymore? TIA for any help or tips!

**Comment by Koenkk** [comment]
See https://github.com/Koenkk/zigbee2mqtt/issues/12721

**Comment by OttoL2** [comment]
Hi, thanks for responding Koen. I did follw the link but my question is a bit more specific... Probably I am just being dense, but I am talking about the Z2M frontend (accessed via the addon to HA). Under "Exposes", only "Device temperature" is shown/exposed for the Xiaomi sensors. That is what will be available via MQTT to HA, right? However, in "State" in that same frontend, BOTH "Device temperature" and "Temperature" are shown. They are NOT the same value for all my sensors. Now, that may be because in the past I added corrections to the the "Temperature" values via "Settings (specific)". But, there (under "Settings (specific)" now you can only adjust "Device temperature". All this started when Z2M last auto-updated a few days back.

I have no idea why "temperature" AND "Device temperature" are shown under "State".  To me, it looks like there is some code/cache/database remnant left whereby the "old" "Temperature" measure still shows up under "State" but nowhere else. Or, I think quite unlikely but certainly you will know, there are 2 temperature sensors, the "Device temperature" AND the ambient "Temperature", and only device temperature is now exposed.

Anyway, not that much of an issue, but it still looks like a - not very consequential - bug?

Best regards,
Maarten

**Reply by Koenkk** [reply]
You will need to manually remove the `temperature` from the `data/state.json` file.

**Comment by OttoL2** [comment]
That did the trick Koen (well at least to remove "temperature" from "state"). Still weird that the "State" values for "Temperature" and "device temperature" where sometimes the same, sometimes different for the 6 sensors I have. Anyway, fixed now. Do you know, by the way, whether this "device temperature" is a separate temp sensor or is it, like in Shelly's, some on-chip temp of the SoC? If on-chip the usefullness is even more limited I reckon... Anyway, many thanks for your help.
Maarten
